spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/ZuulProperties.java

```diff
@@ -109,6 +109,14 @@
 	 * see https://docs.spring.io/spring-security/site/docs/current/reference/html/headers.html#default-security-headers
 	 */
 	private boolean ignoreSecurityHeaders = true;
+	
+	/**
+	 * Flag to force the original query string encoding when building the backend URI in
+	 * SimpleHostRoutingFilter. When activated, query string will be built using
+	 * HttpServletRequest getQueryString() method instead of UriTemplate. Note that this
+	 * flag is not used in RibbonRoutingFilter with eureka discovered services.
```

Comment:
Change to "flag is not used in RibbonRoutingFilter with services found via DiscoveryClient (like Eureka)." 

Since this works with consul, zookeeper etc...